frontend/src/layout/header/UserBox.tsx

```diff
@@ -107,149 +171,189 @@ const WithFloatingMenu: React.FC<{ children: ReactElement }> = ({ children }) =>
     const { t } = useTranslation();
-
-    type State = "main" | "language";
-    const [state, setState] = useState<State>("main");
-
     const user = useUser();
 
-    const items = match(state, {
+    const handleBlur = (event: React.FocusEvent<HTMLDivElement, Element>) => {
+        if (!event.currentTarget.contains(event.relatedTarget as HTMLDivElement)) {
+            close();
+        }
+    };
+
+    const items = match(type, {
         main: () => <>
-            {/* Login button if the user is NOT logged in */}
-            {user === "none" && (
+            <ReturnButton onClick={() => close()}>{t("User features")}</ReturnButton>
+            {isRealUser(user) && <>
                 <MenuItem
-                    icon={<FiLogIn />}
-                    borderBottom
-                    linkTo={CONFIG.auth.loginLink ?? LOGIN_PATH}
-                    htmlLink={!!CONFIG.auth.loginLink}
                     css={{
-                        color: "var(--nav-color)",
                         [`@media not all and (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
-                            display: "none",
+                            borderRadius: "8px 8px 0 0",
                         },
                     }}
-                >{t("user.login")}</MenuItem>
-            )}
-
-            {isRealUser(user) && <>
+                    icon={<FiFolder />}
+                    borderBottom
+                    linkTo="/~manage"
+                    onClick={() => close()}
+                >{t("user.manage-content")}</MenuItem>
                 {user.canUpload && <MenuItem
                     icon={<FiUpload />}
                     linkTo={"/~upload"}
                     onClick={() => close()}
                 >{t("upload.title")}</MenuItem>}
-                <MenuItem
-                    icon={<FiFilm />}
-                    linkTo="/~manage"
-                    onClick={() => close()}
-                >{t("user.manage-content")}</MenuItem>
             </>}
 
-            <MenuItem icon={<HiOutlineTranslate />} onClick={() => setState("language")}>
-                {t("language")}
-            </MenuItem>
-
             {/* Logout button if the user is logged in */}
             {isRealUser(user) && <Logout />}
         </>,
         language: () => <>
-            <MenuItem icon={<FiChevronLeft />} onClick={() => setState("main")} borderBottom>
-                {t("back")}
-            </MenuItem>
-            <LanguageMenu />
+            <ReturnButton onClick={() => close()}>{t("Choose language")}</ReturnButton>
+            <LanguageMenu close={() => close()} />
         </>,
     });
 
-    return (
-        <Floating padding={0}>
+    return <Floating borderWidth={0} padding={0}>
+        <div
+            onClick={e => {
+                if (e.target === e.currentTarget) {
+                    close();
+                }
+            }}
+            onBlur={handleBlur}
```

Comment:
If the `inClick` handler is inlined here, why not the `handleBlur` as well? It's only used here right?

Can you explain what the `onClick` is used for? Also: is the onBlur behavior something one wants? I would rather expect the menu to stay open. But maybe your behavior is better for accessibility?

Reply:
`onClick`: When you open the menus on mobile/tablet, the rest of the page gets greyed out by this div. This is similar (or rather identical) to the burger menu. Without the `onClick`, the menu wouldn't close when that background is clicked (again, inspired by the burger menu).

`onBlur`: This addresses the issue you identified with the floating menus for the SeriesBlock redesign, where you could have multiple menus open at the same time by using tab navigation (this is also true for the newish `addButtons` btw, but I guess it's less of a problem there). I agree that it would be better if the menu stayed open, but I think in that case the focus should be trapped inside them, and I'm not sure how best to achieve that.

I actually think until we figure out a nice way to do that, it would be best to add that `onBlur` functionality to the burger menu as well, as tabbing out of it can have the undesired side effect of scrolling the page while the menu is still open.

Reply:
`onClick`: can you simplify code with `useOnOutsideClick` (from `util/index.tsx`)? Just in case you didn't know that helper.

`onBlur`: mhhh makes sense. Don't have a good idea either. But ye, I agree with the burger menu. We can still improve it in the future.